src/components/Search/server-actions.tsx

```diff
@@ -52,6 +52,59 @@ export interface AskAnswerResult {
     hasAnswer: boolean;
 }
 
+export async function searchSiteContent(args: {
+    query: string;
+    siteSpaceIds?: string[];
+    cacheBust?: string;
+}): Promise<OrderedComputedResult[]> {
+    const { siteSpaceIds, query, cacheBust } = args;
+    const pointer = getContentPointer();
+
+    if (siteSpaceIds?.length === 0) {
+        // if we have no siteSpaces to search in then we won't find anything. skip the call.
+        return [];
+    }
+
+    if ('siteId' in pointer && 'organizationId' in pointer) {
+        const [searchResults, allSiteSpaces] = await Promise.all([
+            api.searchSiteContent(
+                pointer.organizationId,
+                pointer.siteId,
+                query,
+                siteSpaceIds,
+                cacheBust,
+            ),
+            api.getSiteSpaces({
```

Comment:
It looks like it's used only when `!siteSpaceIds` later, so we can skip it here in the other case:

```suggestion
            siteSpaceIds ? null : api.getSiteSpaces({
```